solidity/test/DepositRedemptionTest.js

```diff
@@ -234,15 +230,38 @@ describe("DepositRedemption", async function() {
       expect(tbtcOwed).to.eq.BN(depositValue)
     })
 
-    it("returns SignerFee if we are at-term, caller is TDT holder, and fee is not escrowed", async () => {
+    it("returns zero if we are at-term, caller is TDT holder, and fee is not escrowed", async () => {
       await increaseTime(depositTerm)
+      await feeRebateToken.burn(web3.utils.toBN(testDeposit.address), {
+        from: frtHolder,
+      })
+      const tbtcOwed = await testDeposit.getRedemptionTbtcRequirement.call(
+        tdtHolder,
+      )
+      expect(tbtcOwed).to.eq.BN(new BN(0))
+    })
 
+    it("returns signerFee if we are pre-term, caller is TDT holder and fee is not escrowed", async () => {
+      await feeRebateToken.burn(web3.utils.toBN(testDeposit.address), {
+        from: frtHolder,
+      })
       const tbtcOwed = await testDeposit.getRedemptionTbtcRequirement.call(
         tdtHolder,
       )
       expect(tbtcOwed).to.eq.BN(signerFee)
     })
 
+    it("returns zero if we are in courtesy_call, caller is TDT holder and fee is not escrowed", async () => {
+      await testDeposit.setState(states.COURTESY_CALL)
+      await feeRebateToken.burn(web3.utils.toBN(testDeposit.address), {
+        from: frtHolder,
+      })
+      const tbtcOwed = await testDeposit.getRedemptionTbtcRequirement.call(
+        tdtHolder,
+      )
+      expect(tbtcOwed).to.eq.BN(new BN(0))
```

Comment:
This would also be incorrect, if the above is. Should be signerFee if that’s true.